### PR TITLE
Load donation form pages dynamically

### PR DESCRIPTION
**Pull request by tzhelyazkova**
:heavy_check_mark:  [T224115](https://phabricator.wikimedia.org/T224115)
:page_facing_up: Split the donation form into pages and load them dynamically when `next` or `previous` buttons are clicked. Make `next` button disabled in case of invalid fields.
:memo: Implementing the Overview Component and the Submit functionality will be in separate PRs.

### CODE REVIEW

**Comment by gbirke** [comment]
Wouldn't it make the code simpler and the components more reusable  if the display and event handling of the "previous/next/submit" buttons was handled by `DonationForm`?

**Comment by tzhelyazkova** [comment]
> Wouldn't it make the code simpler and the components more reusable if the display and event handling of the "previous/next/submit" buttons was handled by `DonationForm`?

I looked at the Figma components again and it seems that for Membership the pages are swapped compared to Donation. So yes, it makes more sense to move out the buttons from `Payment` and `Address` so we can reuse them more easily. Thanks :+1:

File: skins/laika/src/components/pages/DonationForm.vue [hunk 1]
```diff
@@ -39,5 +58,67 @@ export default Vue.extend( {
...
+			buttonsVisibility: {
+				previous: false,
+				next: true,
+				submit: false,
+			},
+		};
+	},
+	computed: {
+		currentProperties: {
+			get(): object {
```
Comment (gbirke) [inline skins/laika/src/components/pages/DonationForm.vue]: Another instance where strong typing would both help (by making sure the types match for the component and inside the computed property) and be a pain.

**Comment by tzhelyazkova** [comment]
> Looks good to me. I am not sure if this is still WIP but since Gabriel asked you to create tests for this, it sounded like this is the agreed behavior:
> 
> I think the "Next" button should not be `disabled` when the payment data is not filled out yet. It makes it hard to see why the button is inactive and we should rather make it clickable and show the errors for the missing form values.

I marked it as ready for review so it's not a WIP. I will address Gabriel's comment.
The behavior of the next button is the one described in the ticket so that's a design decision I assume.

**Comment by timEulitz** [comment]
@tzhelyazkova 

> The behavior of the next button is the one described in the ticket so that's a design decision I assume.

I don't think the respective line in the ticket meant that the button would be disabled but rather that the user is shown the error messages and nothing happens. As far as I remember, the button should always be active and validation error messages should be displayed when the button is clicked:

_If a form section is invalid, respective error messages are displayed. Proceeding to the next page is not allowed in that case._

Maybe double-check with Kai or Jan before you implement any tests, I am fairly sure about this but I may be wrong.

**Pull request merged**